### PR TITLE
HPCC-23169 Items in list box is not consistent

### PR DESCRIPTION
**Pull request by miguelvazq**
The grid was not keeping local storage settings that were set previously upon initialization of the grid on the next visit.

Signed-off by: Miguel Vazquez

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/src/ESPUtil.ts [hunk 1]
```diff
@@ -429,13 +429,15 @@ export class UndefinedMemory extends UndefinedMemoryBase {
     }
 }
 
-export function Grid(pagination?, selection?, overrides?, compoundColumns?) {
+export function Grid(pagination?, selection?, overrides?, compoundColumns?, gridName?) {
     var baseClass = [];
     var params = {};
+    var rows = this.LocalStorage.prototype.getItem(gridName, "Number");
```
Comment (GordonSmith) [inline esp/src/src/ESPUtil.ts]: Change to: `Number(localStorage.getItem(gridName))` or `+localStorage.getItem(gridName)`

Comment (GordonSmith) [reply esp/src/src/ESPUtil.ts]: Also when the user changes the "rows" I don't see where you set it in local storage?

Comment (miguelvazq) [reply esp/src/src/ESPUtil.ts]: 1. Will make change and test. 
2. The change of the rows is located in a different call https://github.com/hpcc-systems/HPCC-Platform/blob/17506aeeb78a4c42162dee0e4a3abee49966de7e/esp/src/src/ESPUtil.ts#L306 that was working OK. The reason for this change is that if the user changed it and reloaded the page it did not pull the user preference row number from local storage.

**Review by GordonSmith — APPROVED** [review]
Looks good

**Pull request merged**